### ISSUE TITLE
libselinux does not build with pkgsMusl

### ISSUE THREAD

**Issue by Uthar** [NixOS/nixpkgs#119471]
**Describe the bug**
pkgsMusl.libselinux does not build because of:

```
builder for '/nix/store/bw1k61ngsnhqa9lz2rvrk0y05rka0pkf-libselinux-3.0.drv' failed with exit code 2; last 10 log lines:
  procattr.c:38:14: error: static declaration of ‘gettid’ follows non-static declaration
     38 | static pid_t gettid(void)
        |              ^~~~~~
  In file included from procattr.c:2:
  /nix/store/k515dcqzlwncg8flh97l9dq3vwxn7jvl-musl-1.2.2-dev/include/unistd.h:194:7: note: previous declaration of ‘gettid’ was here
    194 | pid_t gettid(void);
        |       ^~~~~~
  make[1]: *** [Makefile:158: procattr.o] Error 1
  make[1]: Leaving directory '/build/libselinux-3.0/src'
  make: *** [Makefile:44: all] Error 1
[0 built (1 failed)]
error: build of '/nix/store/bw1k61ngsnhqa9lz2rvrk0y05rka0pkf-libselinux-3.0.drv' failed
```

**To Reproduce**
1. `nix repl`
2. `:l https://github.com/NixOS/nixpkgs/archive/6802c2e2ce9ef37976a306ada908e121409093ad.tar.gz`
2. `:b pkgsMusl.libselinux`

**Expected behavior**
The library is built

**Screenshots**

**Additional context**

**Notify maintainers**

@Phreedom 

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.108, NixOS, 20.09.3777.ad7604ddbd9 (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.10`
 - channels(kpg): `""`
 - channels(root): `"nixos-20.09.3777.ad7604ddbd9"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by Uthar** [comment]
fixed in fc141a889d05b05719f6b70f617cc500beea3c38

**Comment by Uthar** [comment]
*fixed on staging, but master is still broken
